### ISSUE TITLE
What can cause different results in communities detected after scaling edge weights of a graph?

### ISSUE THREAD

**Issue by magdalini-anastasiadou** [taynaud/python-louvain#52]
I noticed that if I change all the edge weights in the graph with the same value, community.best_partition doesn't always result to the same communities.

I used the same random state in all cases and the graph is exactly the same just that instead of all edge weights to be equal to 1 for example they may be equal to 5. 
For example, in the example case bellow where I change each weight to 5 the modularity of the communities found is 0.332470647645499 whereas with the communities from the initial graph is 0.334869334679965 and the adjusted rand score = 0.676 between them. 

The definition of modularity cancels out this factor that multiplies the adjacency matrix and as I read about the algorithm I can't find a step that should change the result. What causes that difference?  Is it a numerical error?  Or the method somehow depends on the scale of the values of edge weights?

I asked also at stackoveflow  [https://stackoverflow.com/questions/57560866/what-can-cause-different-results-in-communities-detected-after-scaling-edge-weig](https://stackoverflow.com/questions/57560866/what-can-cause-different-results-in-communities-detected-after-scaling-edge-weig). Below is an example, I've noticed the same behavior in other graphs too.
`

    import networkx as nx
    import community

    from sklearn.metrics import adjusted_rand_score

    def main():
        g = nx.davis_southern_women_graph()
        nodes = g.nodes()
        clusters_init = community.best_partition(g, random_state=10)
        print("modularity with initial clusters = %.15f" % community.modularity(clusters_init, g))

        labels_init = [clusters_init[n] for n in nodes]
        for num in range(1, 9):

            for u, v in g.edges():
                g[u][v]["weight"] = num

            clusters = community.best_partition(g, random_state=10)
            labels = [clusters[n] for n in nodes]
            print("value of edge weight = %d," % num, "modularity = %.15f," % community.modularity(clusters, g),
                  "modularity with initial clusters = %.15f," % community.modularity(clusters_init, g),
                  "adjusted rand score = %.3f" % adjusted_rand_score(labels_pred=labels, labels_true=labels_init))

    if __name__ == "__main__":
        main()
`

**Comment by xiasiyu** [comment]
Anything solutions for this issue?
